hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/HoodieTableSource.java

```diff
@@ -644,10 +647,11 @@ private boolean isPartitioned() {
   }
 
   @VisibleForTesting
-  public Schema getTableAvroSchema() {
+  public HoodieSchema getTableHoodieSchema() {
     try {
       TableSchemaResolver schemaResolver = new TableSchemaResolver(metaClient);
-      return schemaResolver.getTableAvroSchema();
+      Schema avroSchema = schemaResolver.getTableAvroSchema();
+      return HoodieSchema.fromAvroSchema(avroSchema);
```

Comment:
I think we'll have a few PRs all doing this same operation. What do you think about just pushing this down to the TableSchemaResolver so we don't have to modify as much code later?

Reply:
@the-other-tim-brown let me know if this pr:https://github.com/apache/hudi/pull/17456  this aligns with what you said above.